### PR TITLE
VyOS: firewall_interfaces module added

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

##### SUMMARY

Add new resource module vyos_firewall_interfaces.
Targets model - ansible-network/resource_module_models#77
[vyos_firewall_interfaces]

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

vyos_firewall_interfaces
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test/units/modules/network/vyos/test_vyos_firewall_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+                     ])
+            ],
+                state="replaced"))
+        commands = [
+            "delete interfaces ethernet eth0 firewall out name",
+            "delete interfaces ethernet eth0 firewall local name",
+            "delete interfaces ethernet eth2 firewall in name",
+            "delete interfaces ethernet eth2 firewall out name",
+        ]
+        self.execute_module(changed=True, commands=commands)
```
Comment (GomathiselviS) [inline test/units/modules/network/vyos/test_vyos_firewall_interfaces.py:259]: This testcase for 'replaced' state , consists of only deletion commands. Can another testcase be added, which includes addition of rules?

File: test/units/modules/network/vyos/test_vyos_firewall_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+            dict(config=[dict(name='eth0'),
+                         dict(name='eth2')],
+                 state="deleted"))
+        commands = [
+            "delete interfaces ethernet eth0 firewall",
+            "delete interfaces ethernet eth2 firewall"
+        ]
+        self.execute_module(changed=True, commands=commands)
+
+    def test_vyos_firewall_rule_set_03_deleted(self):
```
Comment (GomathiselviS) [inline test/units/modules/network/vyos/test_vyos_firewall_interfaces.py]: Can a comment be added or the name of the testcase be changed, to indicate, what level of granularity of deletion, is tested here ?

**Review by GomathiselviS — CHANGES_REQUESTED** [review]

File: test/units/modules/network/vyos/test_vyos_firewall_interfaces.py (hunk 1 above)
Comment (rohitthakur2590) [reply test/units/modules/network/vyos/test_vyos_firewall_interfaces.py:259]: Done

File: test/units/modules/network/vyos/test_vyos_firewall_interfaces.py (hunk 2 above)
Comment (rohitthakur2590) [reply test/units/modules/network/vyos/test_vyos_firewall_interfaces.py]: Done: I have modified the name of the function/test case keeping the granularity in mind, rest of the scenarios have been taken care in Integration tests, I will cover the remaining in separate PR

File: lib/ansible/module_utils/network/vyos/utils/utils.py [hunk 3]
```diff
@@ -11,9 +11,10 @@
 
 
 def search_obj_in_list(name, lst, key='name'):
```
Comment (NilashishC) [inline lib/ansible/module_utils/network/vyos/utils/utils.py:13]: This method is already defined in network/common/utils.py. Could you remove this from vyos utils and reuse the common one?

File: lib/ansible/modules/network/vyos/vyos_firewall_interfaces.py [hunk 4]
```diff
@@ -0,0 +1,1077 @@
...
+    - gathered
+    default: merged
+"""
+EXAMPLES = """
+# Using merged
+#
+# Before state:
+# -------------
+#
+# vyos@192# run show configuration commands | grep firewall
```
Comment (NilashishC) [inline lib/ansible/modules/network/vyos/vyos_firewall_interfaces.py:117]: Please add an example for updating existing attributes using `state: merged`.

File: lib/ansible/modules/network/vyos/vyos_firewall_interfaces.py [hunk 5]
```diff
@@ -0,0 +1,1077 @@
...
+# set firewall ipv6-name 'V6-LOCAL'
+# set firewall name 'INBOUND'
+# set firewall name 'LOCAL'
+# set firewall name 'OUTBOUND'
+# set interfaces ethernet eth3 firewall 'in'
+# set interfaces ethernet eth3 firewall 'local'
+# set interfaces ethernet eth3 firewall out name 'INBOUND'
+
+
+# Using deleted per interface name
```
Comment (NilashishC) [inline lib/ansible/modules/network/vyos/vyos_firewall_interfaces.py:789]: Is it possible to remove a single firewall rule from an interface?

File: test/integration/targets/vyos_firewall_interfaces/tests/cli/merged.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
+- debug:
+      msg: "START vyos_firewall_interfaces merged integration tests on connection={{ ansible_connection }}"
+
+- include_tasks: _populate_rule_sets.yaml
+
+- include_tasks: _remove_config.yaml
+
+- block:
```
Comment (NilashishC) [inline test/integration/targets/vyos_firewall_interfaces/tests/cli/merged.yaml:9]: Please add a test for updating existing attributes using `state: merged`.

File: test/integration/targets/vyos_firewall_interfaces/tests/cli/deleted.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
+---
+- debug:
+    msg: "Start vyos_firewall_interfaces deleted integration tests ansible_connection={{ ansible_connection }}"
+
+- include_tasks: _populate_rule_sets.yaml
+
+- include_tasks: _populate.yaml
+
+- block:
```
Comment (NilashishC) [inline test/integration/targets/vyos_firewall_interfaces/tests/cli/deleted.yaml:9]: Please add test cases for all possible delete usage.

File: lib/ansible/modules/network/vyos/vyos_firewall_interfaces.py (hunk 5 above)
Comment (rohitthakur2590) [reply lib/ansible/modules/network/vyos/vyos_firewall_interfaces.py:789]: yeah, there is a test case deleted_single added for this.

File: test/integration/targets/vyos_firewall_interfaces/tests/cli/merged.yaml (hunk 6 above)
Comment (rohitthakur2590) [reply test/integration/targets/vyos_firewall_interfaces/tests/cli/merged.yaml:9]: Done

**Review by NilashishC — APPROVED** [review]

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Pull request merged**